plugins/modules/smart_class_parameter_override_value.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# (c) 2023 Evgeni Golov <evgeni@golov.de>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = '''
+---
+module: override_value
+version_added: 3.14.0
+short_description: Manage Smart Class Parameter Override Values
+description:
+  - Manage Smart Class Parameter Override Values
+author:
+  - "Evgeni Golov (@evgeni)"
+options:
+  puppetclass:
+    description:
+    - Puppet Class the Smart Class Parameter belongs to
+    type: str
+    required: true
+  smart_class_parameter:
+    description:
+    - Smart Class Parameter the Override Value belongs to
+    required: true
+    type: str
+  match:
+    description:
+    - Override match
+    required: true
+    type: str
+  omit:
+    description:
+    - Foreman will not send this parameter in classification output
+    required: false
+    type: bool
+  value:
+    description:
+    - Override value, required if omit is false
+    required: false
+    type: raw
+
+extends_documentation_fragment:
+  - theforeman.foreman.foreman
+  - theforeman.foreman.foreman.entity_state
+'''
+
+EXAMPLES = '''
+'''
+
+RETURN = '''
+entity:
+  description: Final state of the affected entities grouped by their type.
+  returned: success
+  type: dict
+  contains:
+    override_values:
+      description: List of override_values.
+      type: list
+      elements: dict
+'''
+
+from ansible_collections.theforeman.foreman.plugins.module_utils.foreman_helper import ForemanEntityAnsibleModule, parameter_value_to_str
+
+
+class ForemanOverrideValueModule(ForemanEntityAnsibleModule):
+    pass
+
+
+def main():
+    module = ForemanOverrideValueModule(
+        foreman_spec=dict(
+            puppetclass=dict(required=True, type='entity', ensure=False),
+            smart_class_parameter=dict(required=True, type='entity', scope=['puppetclass'], search_by='parameter'),
```

Comment:
I am a bit torn on the naming of those two parameters.

We call them `puppetclass_name` and `parameter` in the `smart_class_parameter` module:
https://github.com/evgeni/foreman-ansible-modules/blob/6bf08c69da4bf1d1b877ef0525af4c002d0a74ee/plugins/modules/smart_class_parameter.py#L208-L209

But that seems somehow wrong? We could add aliases for those names, so copy/paste is easier?

Reply:
What seems wrong with it? I feel like the expectation for the parameters that are in both modules would be 1 to 1 naming.

Reply:
I find the `_name` suffix odd, as we use the "name" of an entity everywhere, but this is the only place where it's part of the parameter name.

and `parameter` is - while shorter - not the name the API expects or the UI shows.

Reply:
I would like them to be the same or at least work interchangeably. Will aliasing help? From a UX perspective there's an expectation that they should be the same IMO.

Reply:
Yeah, aliasing would help.

I'd make both modules accept both ways to write it.

Reply:
Great lets do that

Reply:
done